### PR TITLE
fix(keysign): gate personal_sign display decode on strict 0x prefix

### LINKED ISSUE #5402: [Fix] personal_sign Verify screen can display a different message than what actually gets signed
# [Fix] personal_sign Verify screen can display a different message than what actually gets signed

## Problem
`JoinKeysignViewModel.getNormalizedCustomMessage()` feeds `personal_sign` messages through `String.normalizeMessageFormat()`, gated by `isHex()` — a lenient regex (`^(0x)?[0-9A-Fa-f]+$`, the `0x` prefix optional). The actual signing path (`SigningHelper.getKeysignMessages()`) was fixed by PR #4736/#4724 to require a strict `0x` prefix before hex-decoding, but the display path was never updated to match. Any hex-charset message without a leading `0x` (plausible both adversarially and organically — nonces, session tokens, short hex-like strings) displays as decoded text while `SigningHelper` actually signs the raw string bytes — a completely different payload.

## Expected Behavior
The displayed message must always equal the bytes actually signed.

## Steps to Reproduce
1. Receive a `personal_sign` request with message `"56756c7469736967"` (no `0x` prefix, decodes as "Vultisig").
2. The Verify screen displays the decoded text "Vultisig".
3. The actual TSS signature is computed over `keccak256(UTF-8("56756c7469736967"))` — the raw string, not the decoded text.

## Solution
Apply the identical `0x`-prefix gate used in `SigningHelper.kt` to the display path (`StringExtensions.isHex()`/`normalizeMessageFormat()`), or better, have both derive from one shared helper so display and signed bytes can never diverge.

### Must Do
- [ ] Gate `isHex()`/`normalizeMessageFormat()` on a strict `0x` prefix, matching `SigningHelper`'s signing-path check
- [ ] Extract a shared helper both the display and signing paths use, so they cannot drift again

### Must NOT Do
- Don't change the signing digest logic itself (already correct since #4736)
- Don't change custom-message handling for chains other than the hex/EVM-family path

## Acceptance Criteria
- [ ] `./gradlew assembleDebug` succeeds
- [ ] A hex-charset message without a `0x` prefix displays as the literal signed string, not a decoded interpretation
- [ ] New regression test asserts displayed text equals signed content for this exact case

### PR DESCRIPTION
**Pull request by rkokhatskyi**
## Summary

The Verify screen for a `personal_sign` custom-message keysign could display a hex-decoded interpretation of a message that has no `0x` prefix, while `SigningHelper.getKeysignMessages` only hex-decodes a strictly `0x`-prefixed message (fixed on the signing side in #4724/#4736). A bare hex-charset message (e.g. a nonce) displayed as decoded text on the Verify screen while the TSS signature was computed over the raw string bytes — the two paths could diverge.

## Changes

- Extracted `String.isHexPrefixed()` in `StringExtensions.kt` as the single `0x`-prefix gate now shared by `SigningHelper` (replacing its inline `startsWith("0x")`) and `normalizeMessageFormat()` (replacing the lenient `isHex()`, which makes the prefix optional and stays scoped to its other caller, EVM/Tron memo encoding).
- Added a hex-content validity check to `normalizeMessageFormat()`: a `0x`-prefixed remainder containing a non-hex character now falls back to displaying the raw string (same as the existing odd-length fallback), instead of silently decoding garbage bytes into a plausible-looking but wrong string.

## Testing

- `./gradlew :data:testDebugUnitTest` — full suite green, including the existing `SigningHelperCustomMessageTest` (unaffected) and two new test files: `StringExtensionsTest` (the display-path gate and its edge cases) and `PersonalSignDisplayMatchesSignedMessageTest` (cross-path invariant: hashing the UTF-8 bytes of what's displayed reproduces exactly what `SigningHelper` signs).
- `./gradlew :app:testDebugUnitTest` — green (unaffected by this change).
- `./gradlew :data:ktfmtCheckMain :data:ktfmtCheckTest` — clean.

Closes #5402

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved message handling to consistently distinguish hexadecimal content from plain text.
  * Prevented malformed or ambiguous hexadecimal messages from being incorrectly decoded.
  * Ensured displayed personal-sign messages match the content used for signing.

* **Tests**
  * Added coverage for prefixed, unprefixed, malformed, empty, and valid hexadecimal message formats.

### CODE REVIEW

File: data/src/main/kotlin/com/vultisig/wallet/data/common/StringExtensions.kt [hunk 1]
```diff
@@ -43,11 +47,21 @@ fun String.toByteStringOrHex(): ByteString {
...
     return try {
-        if (this.isHex()) {
+        if (this.isHexPrefixed()) {
             val hex = this.remove0x()
-            if (hex.length % 2 != 0) {
+            // A malformed remainder (odd length, or a non-hex character) must fall back to the
+            // raw string rather than decode: SigningHelper's hex-decode silently maps invalid
+            // digits to garbage bytes, so decoding here too could display a clean-looking (but
+            // wrong) string for content that doesn't actually represent well-formed hex.
+            if (hex.length % 2 != 0 || !hex.all { it.digitToIntOrNull(16) != null }) {
```
[the remark this answers is not in the document]
Comment (rkokhatskyi) [reply data/src/main/kotlin/com/vultisig/wallet/data/common/StringExtensions.kt:64]: Declining this one: I checked iOS's `Data(hex:)` (CryptoSwift, `Array<UInt8>.init(hex:)`) and Windows's `Buffer.from(stripHexPrefix(message), 'hex')` for a malformed-hex input like "0x008z41" — iOS wipes to empty bytes on the invalid char, Windows truncates to the first valid byte pair, and Android's current lenient `Character.digit`-based decode produces yet another byte sequence, but all three stay on the hex-decode branch rather than falling back to raw UTF-8. Making SigningHelper reject malformed hex and sign raw UTF-8 instead would flip Android onto a fourth, different branch, reintroducing the exact cross-platform digest/message-ID mismatch PR #4736 fixed — and the ticket explicitly excludes changing signing digest logic. The three-way byte-level disagreement on genuinely malformed hex is a separate, pre-existing gap across all platforms, not something this PR's scope (the missing-0x-prefix display bug) should touch. The test exclusion is intentional for the same reason — malformed-hex fallback is already covered directly in StringExtensionsTest (odd-length and non-hex-char cases), just not folded into this specific display-equals-signed-bytes invariant test since that invariant doesn't hold for inputs where signing stays lenient while display conservatively falls back.

**Review by johnnyluo — APPROVED** [review]
LGTM
